docs/token-contracts/fa12/4-fa12-archetype.md

```diff
@@ -0,0 +1,447 @@
+---
+id: 4-fa12-archetype
+title: Deploy a Verified Implementation of FA1.2 Contract
+sidebar_label: FA1.2 Archetype
+---
+
+## Introduction
+This article presents the procedure to deploy a verified implementation of the FA1.2 contract.
+
+[Program verification](https://en.wikipedia.org/wiki/Formal_verification) is a technique to mathematically prove that a program has the required properties. It provides a high level of confidence that the contract executes correctly according to its specification.
+
+The formal properties of the FA1.2 contract are presented [below](#formal-properties).
+
+The contract and its formal specification are written in the [Archetype language](https://archetype-lang.org/), a high-level domain-specific language dedicated to the development of verified smart contracts for the Tezos blockchain.
+
+For verification purposes, Archetype translates the contract to the Why3 language; [Why3](http://why3.lri.fr/) is a generic program verification platform; it generates proof obligations and calls [SMT](https://en.wikipedia.org/wiki/Satisfiability_modulo_theories) solvers to solve them.
+
+## Generate Michelson
+The Archetype source code is available [here](https://github.com/edukera/archetype-lang/blob/0a5ad0832709ac102a14534f22d4f94cb185866d/contracts/fa12.arl).
+
+> SHA256 (fa12.arl) = d38260e5f55e630880ecd5d5ed32eb91988d2f34b43a7c031275d8f8a8715184
+
+The following command generates the Michelson version of the contract in the `fa12.tz` file:
+```
+$ archetype fa12.arl > fa12.tz
```

Comment:
I didn't see a specified/min/max Archetype version in the `fa12.arl`.
- Is there a minimum version? If I had issues, I might try that exact commit
- Is Archetype sufficiently stable that future releases are unlikely to break this contract?

My build failed:
```bash
$ archetype fa12.arl
File "fa12.arl", line 6, characters 5-6: lexical error: unexpected char: !
Fatal error: exception Archetype.Lexer.LexError
$ archetype --version 
0.1.10
```

Even when I upgraded to `archetype 1.2.0`:
```bash
$ archetype fa12.arl
File "fa12.arl", line 6, characters 5-6: lexical error: unexpected char: !Fatal error: exception Archetype.Lexer.LexError
$ archetype --version 
1.2.0
```

Reply:
Hello Michael,
I have added in the doc the minimum required version, that is 1.2.0.
The archetype syntax is stable and future version should not break it.
I do not reproduce the error you get with version 1.2.0.
I tried the following and it works:
```
cd /tmp
$ mkdir archetype
$ cd archetype
$ opam update
$ opam switch create . 4.07.0
$ eval $(opam env)
$ opam install archetype --yes
$ curl -o fa12.arl -s https://raw.githubusercontent.com/edukera/archetype-lang/0a5ad0832709ac102a14534f22d4f94cb185866d/contracts/fa12.arl
$ archetype fa12.arl
```

Reply:
Hello Michael, have you had the opportunity to run the commands above?

Reply:
Hi, yes: that worked with one change, I had to run `opam update` with `sudo`

Reply:
Hello Michael, just checking: are you waiting for any change from our side?